engine/src/main/java/org/kigalisim/engine/state/OverridingConverterStateGetter.java

```
@@ -1,136 +1,294 @@
  *
- * <p>This interface defines methods for retrieving and overriding various
- * state values needed for unit conversions and calculations in the engine.</p>
+ * <p>This class wraps an inner StateGetter and allows individual values to be
+ * overridden while delegating to the inner getter for non-overridden values.</p>
  */
-public interface OverridingConverterStateGetter {
+public class OverridingConverterStateGetter implements StateGetter {
+
+  private final StateGetter innerGetter;
+  private Optional<EngineNumber> energyIntensity = Optional.empty();
+  private Optional<EngineNumber> amortizedUnitVolume = Optional.empty();
+  private Optional<EngineNumber> population = Optional.empty();
+  private Optional<EngineNumber> yearsElapsed = Optional.empty();
+  private Optional<EngineNumber> totalConsumption = Optional.empty();
+  private Optional<EngineNumber> energyConsumption = Optional.empty();
+  private Optional<EngineNumber> volume = Optional.empty();
+  private Optional<EngineNumber> amortizedUnitConsumption = Optional.empty();
+  private Optional<EngineNumber> populationChange = Optional.empty();
+  private Optional<EngineNumber> substanceConsumption = Optional.empty();
+
+  /**
+   * Create a new overriding converter state getter.
+   *
+   * @param innerGetter The base state getter to wrap
+   */
+  public OverridingConverterStateGetter(StateGetter innerGetter) {
+    this.innerGetter = innerGetter;
+  }
+
+  /**
+   * Set total values for different stream types.
+   *
+   * @param streamName The name of the stream (sales, manufacture, import, etc.)
+   * @param value The value to set for the stream
+   */
+  public void setTotal(String streamName, EngineNumber value) {
+    switch (streamName) {
+      case "sales", "manufacture", "import" -> setVolume(value);
+      case "equipment", "priorEquipment" -> setPopulation(value);
+      case "consumption" -> setConsumption(value);
+      default -> {
+        // No-op for unrecognized stream names
```

Please throw an exception for unrecognized